src/main/java/core/basesyntax/service/converter/DataConverterImpl.java

```
@@ -0,0 +1,32 @@
+package core.basesyntax.service.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String SPLIT = ",";
+    private static final int INDEX_TYPE = 0;
+    private static final int INDEX_FRUIT = 1;
+    private static final int INDEX_NUMBER = 2;
+
+    @Override
+    public List<FruitTransaction> convertAll(List<String> line) {
+        return line.stream()
+                .skip(1)

```

The code unconditionally calls `.skip(1)` to skip a header. The requirements say to ensure header presence/correctness (at least ensure the first line is not treated as data). If the file has no header or the first line is malformed this will silently skip needed data or mis-handle input. Validate that the first line is a header (e.g., equals/contains `type,fruit,quantity`) or safely handle both cases. Quote from description: "CSV header presence and correctness (at least ensure first line is not treated as data)."